### PR TITLE
[Docs] Convert `notes/cuda.rst` from rST to MyST Markdown

### LINKED ISSUE #182483: [Docathon] Convert `notes/cuda.rst` from rST to MyST Markdown
## Description

Convert [`docs/source/notes/cuda.rst`](https://github.com/pytorch/pytorch/blob/main/docs/source/notes/cuda.rst) from reStructuredText to MyST Markdown and ensure that doc tests pass.

## Steps

1. **Assign the issue** to yourself using the `/assigntome` comment.

2. **Rename the file using `git mv`:**
   ```bash
   git mv docs/source/notes/cuda.rst docs/source/notes/cuda.md
   ```
   **Important!** Use `git mv` to convert the file — DO NOT delete the `.rst` and create a new `.md`.
   We won't accept PRs with deleted `.rst` files. We want to preserve the git log.

3. **Convert the content** from rST to MyST Markdown syntax.

4. **Build the docs locally** as described in the [README](https://github.com/pytorch/pytorch/tree/main/docs).
   Preview locally with `make serve` to ensure the HTML renders correctly.

5. **Run lintrunner** from the root of the repository:
   ```bash
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

6. **Submit a PR** with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:
   - `pull / linux-docs / build-docs-python-false (pull_request)`
   - `linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)`

   Doc tests start after the Docker build, which might take about 30 minutes.
   The doc build itself should take about 25 minutes. When the build is done,
   you can see the preview of your build by clicking
   **📄 Preview Python docs built from this PR** in the pytorch-bot comment.

7. **Label your PR** with: `docathon-2026`, `module: docs`.

8. **Compare the preview** of the pages you converted with the same pages on the main branch.
   The converted pages in preview must look the same as in the main branch.

9. Once the PR is accepted, one of the maintainers will merge it.

## References

- [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown) (wiki)
- [Converter script](https://github.com/svekars/sphinx-read-thedocs-test/blob/main/rst_to_myst_md_converter.py)
- [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html)
- [Example PR](https://github.com/pytorch/pytorch/pull/154438)

cc @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by XAheli**
Fixes #182483

### Changes
- Renamed `cuda.rst` → `cuda.md` via `git mv`
- Converted `.. meta::` directive to YAML frontmatter
- Converted all 21 cross-reference labels (`.. _label:` → `(label)=`)
- Converted RST headers (`=`, `-`, `^`, `~` underlines) to Markdown headings (`#`–`####`)
- Converted 48 code blocks (`.. code::` / `.. code-block::` / `::`) to fenced code blocks
- Converted 8 admonitions (`.. note::`, `.. warning::`) to MyST `:::{note}` / `:::{warning}` directives
- Converted RST roles (`:mod:`, `:class:`, `:func:`, `:meth:`, `:ref:`, `:any:`, `:attr:`) to MyST syntax (`{role}`)
- Converted 14 external link definitions to inline Markdown links
- Converted RST inline code (``` ``text`` ```) to Markdown (`` `text` ``)
- `.. cublas-workspaces:` (line 732 in original — a comment missing the `_` prefix, not a label) converted to `<!-- cublas-workspaces -->` HTML comment; open to making it a proper `(cublas-workspaces)=` label if it was intended as one

### Validation
- `lintrunner -a`: zero issues
- Full `make html` build: completed successfully; all 22 warnings are pre-existing `autosummary` failures in `fx.experimental.md` (z3 imports) — **zero warnings from `cuda.md`**
- Generated HTML (`notes/cuda.html`, 452 KB) renders correctly with all labels intact

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

**Comment by svekars** [comment]
sanity check is failing - the change is too large. Do this: undo all your changes. git mv .rst .md and make a commit. Then apply conversion changes and commit again. Then push. This should preserve the git log and you should have fewer changed lines.

**Comment by XAheli** [comment]
Hi @svekars I've restructured into 2 commits (pure git mv rename + conversion) and added the missing `redirects.py` entry. The doc-redirects-check should now pass. For pr-sanity-checks, the RST-to-MyST format conversion inherently exceeds 2000 LOC even with git rename detection (2159 LOC). Could you add the skip-pr-sanity-checks label?

File: docs/source/redirects.py [hunk 1]
```diff
@@ -159,4 +159,6 @@
         "user_guide/torch_compiler/compile/"
         "programming_model.where_to_apply_compile.html"
     ),
+    # Auto-generated redirects for moved files
+    "notes/cuda": "notes/cuda.html",
```
Comment (svekars) [inline docs/source/redirects.py]: not, needed, failing link check can be ignored: 
```suggestion
```

**Comment by svekars** [comment]
git log seem to be still lost: https://docs-preview.pytorch.org/pytorch/pytorch/183426/notes/cuda.html

File: docs/source/redirects.py (hunk 1 above)
Comment (XAheli) [reply docs/source/redirects.py]: fixed it!

**Comment by XAheli** [comment]
> git log seem to be still lost: https://docs-preview.pytorch.org/pytorch/pytorch/183426/notes/cuda.html

the history is actually preserved imo. The first commit is a pure `git mv` (100% similarity), second commit does the conversion. `git log --follow docs/source/notes/cuda.md` traces through the rename correctly after merge. GitHub's "Files changed" tab just doesn't show it as a rename because the overall content diff is too large for its detection threshold ig? The actual git history is fine though.

**Comment by svekars** [comment]
In the HTML at the top, there is Created On date: [docs-preview.pytorch.org/pytorch/pytorch/183426/notes/cuda.html](https://docs-preview.pytorch.org/pytorch/pytorch/183426/notes/cuda.html) - it shows May 12 2026 and the pr sanity check still failing.

**Comment by XAheli** [comment]
> In the HTML at the top, there is Created On date: [docs-preview.pytorch.org/pytorch/pytorch/183426/notes/cuda.html](https://docs-preview.pytorch.org/pytorch/pytorch/183426/notes/cuda.html) - it shows May 12 2026.

```
cd /workspaces/pytorch-devcontainers/pytorch && echo "=== Full follow history ===" && git log --follow --oneline -- docs/source/notes/cuda.md | tail -5 && echo "=== Created date ===" && git log --follow --format="%ad" --date=format:"%b %d, %Y" -- docs/source/notes/cuda.md | tail -1 && echo "=== Last updated ===" && git log -1 --format="%ad" --date=format:"%b %d, %Y" -- docs/source/notes/cuda.md
=== Full follow history ===
88275da5e87 CUDA documentation tweaks (#858)
b87c113cf4f CUDA documentation enhancement and docs versioning (#848)
a38749d15f9 Fix cuda notes
4cc11066b26 Add torch.utils.data docs and improve notes (#460)
15c1dad3408 Minor fixes and torch.cuda docs
=== Created date ===
Jan 16, 2017
=== Last updated ===
May 13, 2026
```
seeing this locally! also the following in [weblink](https://docs-preview.pytorch.org/pytorch/pytorch/183426/notes/cuda.html)

Screenshot From 2026-05-15 14-48-26



> the pr sanity check still failing.

Imo the pr-sanity-check is expected to fail here. The RST-to-MyST conversion exceeds [2000 line limit](https://github.com/pytorch/pytorch/blob/main/.github/scripts/pr-sanity-check.sh#L51). I don't think there's way to reduce it further since it's a format conversion of ~1800 line file (please correct me if wrong)?

**Comment by svekars** [comment]
Great! git log is preserved. Can you undo indentations in code blocks? I tried here and it looks better and would pass sanity check: https://github.com/pytorch/pytorch/pull/183925. You can submit a follow up PR to fix indentations for extra points.

**Comment by XAheli** [comment]
> Great! git log is preserved. Can you undo indentations in code blocks? I tried here and it looks better and would pass sanity check: #183925. You can submit a follow up PR to fix indentations for extra points.

Hi @svekars updated with the minimal approach matching your example in #183925. Will open a followup PR for the indentation cleanup once this merges.

File: docs/source/notes/cuda.md [hunk 2]
```diff
@@ -1785,15 +1747,13 @@ or if they'll run concurrently, passing them as a tuple to a single invocation o
 :func:`~torch.cuda.make_graphed_callables` is not allowed. Instead, you must call
 :func:`~torch.cuda.make_graphed_callables` separately for each one.
 
-.. _graph-data-dependent-control-flow:
```
Comment (svekars) [inline docs/source/notes/cuda.md:1788]: need to add that back: /__w/pytorch/pytorch/docs/source/user_guide/torch_compiler/torch.compiler_cudagraph_trees.md:13: WARNING: undefined label: 'graph-data-dependent-control-flow'

see [log](https://github.com/pytorch/pytorch/actions/runs/25955101187/job/76338308867?pr=183426)

Comment (XAheli) [reply docs/source/notes/cuda.md:1788]: addressed in the latest commit

**Comment by XAheli** [comment]
Followup here #184112 since deadline is approaching!

cc @svekars

**Comment by XAheli** [comment]
Hi @svekars could you please merge this? I'll then remove the prior commits from the followup. Thanks!

**Comment by svekars** [comment]
ok, this looks good for the first iteration. Check the preview here: https://docs-preview.pytorch.org/pytorch/pytorch/183426/notes/cuda.html - in your follow up PR, make sure you fix :ref:, :mod:, and other Sphinx directives as well as indendtation.

**Review by svekars — APPROVED** [review]
